src/stdune/console.mli

```diff
@@ -41,6 +41,17 @@ include Backend.S
     {[ print_user_message (User_message.make paragraphs) ]} *)
 val print : User_message.Style.t Pp.t list -> unit
 
+(** [lock] captures any messages written to the console, which are rendered in
+    the previous console mode upon being [unlock]ed.
+
+    To allow these to be used alongside [Fiber] jobs, we expose this unsafe
+    interface instead of [with_lock: (unit -> unit) -> unit]. Double-[lock]ing
+    or [unlock]ing the console will lead to output errors (no output, repeated
+    output, etc). *)
```

Comment:
Can we just forbid double locking/unlocking via runtime checks?

Reply:
This was also my idea, but I wasn't sure what to do in the case of composing a buffered terminal with a non-buffered terminal.

Reply:
I spoke to @jeremiedimino about the behavior of terminal locking in the presence of composed terminals, and we concluded that we don't need to worry about it for now, so I've adjusted the interface.